onnxruntime/core/providers/coreml/builders/impl/activation_op_builder.cc

```diff
@@ -54,6 +74,67 @@
 
 // Operator support related
 
+namespace {
+// assumes that node.OpType() == "PRelu"
+bool IsPReluOpSupported(const Node& node, const OpBuilderInputParams& input_params,
+                        const logging::Logger& logger) {
+  const auto& input_defs = node.InputDefs();
+
+  // X input rank must be at least 3
+  {
+    std::vector<int64_t> x_shape;
+    if (!GetShape(*input_defs[0], x_shape, logger)) {
+      return false;
+    }
+    if (x_shape.size() < 3) {
+      LOGS(logger, VERBOSE) << "PRelu 'X' input must have at least 3 dimensions";
+      return false;
+    }
+  }
+
+  // slope input must be constant
+  {
+    const auto& initializers = input_params.graph_viewer.GetAllInitializedTensors();
+    const auto initializer_it = initializers.find(input_defs[1]->Name());
+    if (initializer_it == initializers.end()) {
+      LOGS(logger, VERBOSE) << "PRelu 'slope' input must be an initializer tensor";
+      return false;
+    }
+  }
```

Comment:
Should this be checking constant initializers? 

Unlikely, but in theory the slope could be an initializer that can be overridden via a graph input.

Reply:
good point. I'll fix this one in this PR, but I think there are other places that may need to be similarly updated.